### ISSUE TITLE
Consistently Generate Random Data

### ISSUE THREAD

**Issue by jeremy-audet** [SatelliteQE/robottelo#1378]
TL;DR Let's use FauxFactory everywhere.

Currently, we use three different tools for generating random data throughout the codebase:
- Our internal `generate_*` methods.
- The `orm.<Type>Field.get_value` methods.
- `FauxFactory.generate_*` methods.

This is madness:
- A contributor has to learn as many as _three_ different tools for generating data.
- The project has an inherently higher maintenance burdern. For example, let's say we want to follow [RFC 2606](https://www.rfc-editor.org/rfc/rfc2606.txt) when generating domain names. In this case, both an internal `generate_*` method and `FauxFactory.generate_url` need to be updated. It would be bettter if we could update just one method.

As of late, we have been moving to `orm.<Type>Field`. However, I do not think this is the best solution.
- Each field is basically a wrapper around a FauxFactory method. Each field offers a little bit of custom logic stating, for example, how long a field should be or which types of characters should be generated. They have been created because the API layer needs a set of classes to implement an ORM-like abstraction. This allows e.g. the `create()` method to ask each entity field for customized random data in a consistent way, and it allows for some useful type checking.
- When `orm.<Type>Field` is used in a test, the user generally knows exactly what they need. The user might know that they want 8 random alphanumeric characters. Thus, FauxFactory or `orm.<Type>Field` could work equally well... right? Well, no. It turns out that not all `orm.<Type>Field` classes are capable of generating random data. For example, `orm.DateField.get_value()` fails. Also, not all classes wrap FauxFactory's equivalent very well. For example, the amount of flux in `orm.StringField`'s interface demonstrates this.
- The `orm.<Type>Field` classes are really intended to solve a problem specific to the API layer, and the API layer is not done evolving. There are several improvements I'd like to make to the API layer, and "improvements" means "breakage". If the `orm.<Type>Field` classes are used pervasively, then this will be much harder to do.
- The `orm.<Type>Fields` are project-specific code. Maintenance burden. :(

I think FauxFactory is the best tool for our needs. Its interface is more stable, it offers a greater variety of methods for generating data, those methods have a richer set of options, and it is not project-specific.
